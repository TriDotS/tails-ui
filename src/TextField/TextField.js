import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledInput = styled.input`
  &:focus {
    outline: none;
  }
`

const TextField = ({
  label,
  value,
  type,
  name,
  placeholder,
  color,
  ...props
}) => (
  <div>
    {label && (
      <label className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
        {label}
      </label>
    )}
    <StyledInput
      className={`bg-grey-lighter appearance-none border-2 border-grey-lighter hover:border-${color} rounded w-full py-2 px-4 text-grey-darker`}
      type={type}
      value={value}
      name={name}
      placeholder={placeholder}
      {...props}
    />
  </div>
)

TextField.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  color: PropTypes.string
}

export default TextField
