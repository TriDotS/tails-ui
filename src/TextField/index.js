import React from 'react'
import styled from 'styled-components'

const TextField = styled.input`
  width: ${props => (props.fullWidth ? '100%' : null)};
  padding: 10px 10px;
  box-sizing: border-box;
  border: 1px solid silver;
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  font-size: 1em;
  transition: 0.5s;
  outline: none;

  &:focus {
    border: 1px solid grey;
  }

  &:invalid {
    border: 1px solid indianred;
  }
`

export default ({ type, placeholder, labelText, rounded, fullWidth }) =>
  <div>
    {labelText && <label>{labelText}</label>}
    <TextField
      type={type}
      rounded={rounded}
      fullWidth={fullWidth}
      placeholder={placeholder || null}
    />
  </div>
