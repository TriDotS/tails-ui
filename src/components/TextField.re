let component = ReasonReact.statelessComponent("TextField");

let make =
    (
      ~label,
      ~value,
      ~htmlType,
      ~name,
      ~placeholder,
      ~color,
      ~hasError,
      _children
    ) => {
  ...component,
  render: _self => {
    let errorStyle =
      [
        Js.Boolean.to_js_boolean(hasError) === Js.Boolean.to_js_boolean(true) ?
          "border-red hover:border-red" :
          {j|border-grey-light hover:border-$(color)|j}
      ]
      |> String.concat("");
    <div>
      (
        label !== "" ?
          <label
            className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
            (ReasonReact.stringToElement(label))
          </label> :
          ReasonReact.nullElement
      )
      <input
        className={j|bg-grey-lighter appearance-none border-2 rounded w-full py-2 px-4 text-grey-darker $(errorStyle)|j}
        _type=htmlType
        value
        name
        placeholder
      />
    </div>;
  }
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~label=jsProps##label,
      ~value=jsProps##value,
      ~htmlType=jsProps##htmlType,
      ~name=jsProps##name,
      ~placeholder=jsProps##placeholder,
      ~color=jsProps##color,
      ~hasError=jsProps##hasError,
      [||]
    )
  );
