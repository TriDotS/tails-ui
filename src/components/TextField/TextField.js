import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledInput = styled.input`
  &:focus {
    outline: none;
  }
`

const TextField = ({ label, value, type, name, placeholder }) => (
  <div>
    {label && (
      <label className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
        {label}
      </label>
    )}
    <StyledInput
      className="bg-grey-lighter appearance-none border-2 border-grey-lighter hover:border-grey rounded w-full py-2 px-4 text-grey-darker"
      type={type}
      value={value}
      name={name}
      placeholder={placeholder}
    />
  </div>
)

TextField.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string
}

export default TextField
