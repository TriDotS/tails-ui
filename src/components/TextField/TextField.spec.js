import React from 'react'
import TextField from './TextField'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<TextField />, div)
})

it('generate text field snapshot', () => {
  const component = renderer.create(<TextField />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
