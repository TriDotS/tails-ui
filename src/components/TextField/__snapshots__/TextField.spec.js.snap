// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate text field snapshot 1`] = `
<div
  className="sc-bdVaJa fPPamc"
>
  <input
    className="sc-bwzfXH dHeuUr"
    placeholder={null}
    type={undefined}
  />
</div>
`;
